sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java

```diff
@@ -178,8 +217,24 @@ public synchronized void setMetrics(
   @SuppressWarnings("NullAway")
   public synchronized void stop() {
     if (isFrameMetricsAggregatorAvailable()) {
-      frameMetricsAggregator.stop();
-      frameMetricsAggregator.reset();
+      try {
+        if (MainThreadChecker.isMainThread()) {
+          frameMetricsAggregator.stop();
+          frameMetricsAggregator.reset();
+        } else {
+          handler.post(
+              () -> {
+                try {
+                  frameMetricsAggregator.stop();
+                  frameMetricsAggregator.reset();
```

Comment:
`reset` can be out of both blocks, it does not touch `OnFrameMetricsAvailableListener` and it copies the instance of `mMetrics`